insights/parsers/tomcat_xml.py

```diff
@@ -0,0 +1,76 @@
+"""
+tomcat_xml - XML files for Tomcat
+=================================
+
+Classes to parse Tomcat XML configuration files.
+
+.. note::
+    The tomcat XML files are gotten from the directory specified in the java
+    commands
+
+"""
+from .. import parser, XMLParser
+
+
+@parser("tomcat_web.xml")
+class TomcatWebXml(XMLParser):
+    """
+    Parse the `web.xml` of Tomcat.
+
+    Currently it only stores the setting of `session-timeout`.
+
+    Attributes:
+        data (dict): special settings, e.g. ``session-timeout`` get from the
+            xml file.
+
+    Sample input::
+
+        <?xml version="1.0" encoding="ISO-8859-1"?>
+        <web-app xmlns="http://java.sun.com/xml/ns/javaee"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://java.sun.com/xml/ns/javaee http://java.sun.com/xml/ns/javaee/web-app_2_5.xsd"
+            version="2.5">
+
+            <servlet>
+                <servlet-name>default</servlet-name>
+                <servlet-class>org.apache.catalina.servlets.DefaultServlet</servlet-class>
+                <init-param>
+                    <param-name>debug</param-name>
+                    <param-value>0</param-value>
+                </init-param>
+                <init-param>
+                    <param-name>listings</param-name>
+                    <param-value>false</param-value>
+                </init-param>
+                <load-on-startup>1</load-on-startup>
+            </servlet>
+
+            <session-config>
+                <session-timeout>30</session-timeout>
+            </session-config>
+
+            <welcome-file-list>
+                <welcome-file>index.html</welcome-file>
+                <welcome-file>index.htm</welcome-file>
+                <welcome-file>index.jsp</welcome-file>
+            </welcome-file-list>
+
+        </web-app>
+
+    Examples:
+        >>> type(web_xml)
+        <class 'insights.parsers.tomcat_xml.TomcatWebXml'>
+        >>> web_xml.get('session-timeout') == 30
+        True
+    """
+
+    TIMEOUT_KEYWORD = ''
+
+    def _parse_dom(self):
```

Comment:
Before we start using this method, I think we should consider whether it should be a public method in the base class.  Currently its private so its not included in any of the documentation and a user would not know that its there and could be implemented in a child class.  Now that I understand how its used I think the base class should be changed to make this a public method, and the documentation should be clear about how to use it.

Reply:
Thanks @bfahr , that makes sense to us. We'll make it public.